crates/matrix-sdk-crypto/src/session_manager/group_sessions/share_strategy.rs

```diff
@@ -1340,7 +1340,12 @@ mod tests {
                 .with_dehydrated_device(bob_dehydrated_device_id, true)
                 .build_response();
             allow_duplicates! {
-                with_settings!({sort_maps => true}, { assert_json_snapshot!(ruma_response_to_json(keys_query.clone())) });
+                with_settings!({sort_maps => true}, {
+                    assert_json_snapshot!(
+                        "should_share_with_verified_dehydrated_device",
```

Comment:
ok, so the problem here is that you haven't actually changed anything: this is the default name for the snapshot.

I think, rather than overriding the snapshot name, you actually want to disable [`prepend_module_to_snapshot`](https://docs.rs/insta/latest/insta/struct.Settings.html#method.prepend_module_to_snapshot). I don't really know why insta includes the module name in the snapshot name, since it's implied by the directory name.

You can change the value with `with_settings!({prepend_module_to_snapshot => false}, /* ... */)`.

Reply:
Frankly I followed [your previous PR](https://github.com/matrix-org/matrix-rust-sdk/pull/4539), because it fixed the similar issue.

Reply:
Tried to disable `prepend_module_to_snapshot`, but snapshot file was not generated.
Pushed [commit](https://github.com/matrix-org/matrix-rust-sdk/pull/4625/commits/e9b7aefaf61396127b41cfcfaa09575dc3f6739b).

Reply:
you've disabled `sort_maps` rather than `prepend_module_to_snapshot`